proof/refine/RISCV64/Schedule_R.thy

```diff
@@ -4192,90 +4025,87 @@ lemma getRefillFull_sp:
   apply (fastforce simp: obj_at'_def refillSize_def)
   done
 
+(* FIXME RT: move to AInvs *)
+lemma get_refill_tail_sp:
+  "\<lbrace>P\<rbrace>
+   get_refill_tail sc_ptr
+   \<lbrace>\<lambda>rv s. P s \<and> (\<exists>sc n. ko_at (Structures_A.SchedContext sc n) sc_ptr s \<and> rv = refill_tl sc)\<rbrace>"
+  apply (wpsimp wp: get_refill_tail_wp)
+  apply (clarsimp simp: obj_at_def)
+  done
+
+lemma getRefillTail_corres:
+  "sc_ptr = scPtr \<Longrightarrow>
+   corres (\<lambda>rv rv'. refill_map rv' = rv)
+     (valid_objs and pspace_aligned and pspace_distinct
+      and is_active_sc sc_ptr and sc_refills_sc_at (\<lambda>refills. refills \<noteq> []) sc_ptr)
+     valid_objs'
+     (get_refill_tail sc_ptr) (getRefillTail scPtr)"
+  apply (clarsimp simp: get_refill_tail_def getRefillTail_def read_refill_tail_def
+                        readRefillTail_def getSchedContext_def[symmetric]
+                        read_sched_context_get_sched_context
+                        readSchedContext_def getObject_def[symmetric])
```

Comment:
dunno if this is from older proofs, but checking that you know you can do `clarsimp simp: ... simp flip: ...` to not need the `[symmetric]`